src/WebJobs.Script.Analyzers/WebJobsAttributeAnalyzer.cs

```diff
@@ -0,0 +1,141 @@
+﻿using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Azure.Functions.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class WebJobsAttributeAnalyzer : DiagnosticAnalyzer
+    {
+        // TODO: Scope this to per-project
+        JobHostMetadataProvider _tooling;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(DiagnosticDescriptors.IllegalFunctionName); } }
+
+    public static void VerifyWebJobsLoaded() 
+        {
+            var x = new JobHost(new OptionsWrapper<JobHostOptions>(new JobHostOptions()), null);
+        }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            // https://stackoverflow.com/questions/62638455/analyzer-with-code-fix-project-template-is-broken
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+
+            VerifyWebJobsLoaded();
+
+            // Analyze method signatures.
+            context.RegisterSyntaxNodeAction(AnalyzeMethodDeclarationNode, SyntaxKind.MethodDeclaration);
+
+            // Hook compilation to get the assemblies' references and build the WebJob tooling interfaces.
+            context.RegisterCompilationStartAction(AnalyzeCompilation);
+        }
+
+        private void AnalyzeCompilation(CompilationStartAnalysisContext context)
+        {
+            var compilation = context.Compilation;
+
+            AssemblyCache.Instance.Build(compilation);
+            _tooling = AssemblyCache.Instance.Tooling;
```

Comment:
Could we document what are the intentions of this method/why we are doing what we are doing here, to help future readers ? Also, for a present reader😁 I am lost here 😕

Reply:
Removed `AssemblyCache` for now after conversation offline; will reintroduce in the PR for the analyzer that actually makes use of it.